### DISCUSSION TITLE
Telemetry channel log filtering

### DISCUSSION THREAD

**Discussion by jhsu3** [nasa/fprime#2800] [Q&A]
In GDS, all telemetry gets logged to channel.log. Is there a way to filter logging of desired telemetry to different files?

For example, filter a component's telemetry to component.log

**Comment by thomas-bc** [comment]
This is not a capability the F´ GDS offers at this time, but could make for a feature request.

 Note that you could also [write a custom plugin](https://nasa.github.io/fprime/HowTo/develop-gds-plugins.html) that would register a ChannelDecoder, filter on your component and do whatever you'd like with the data.
We have an example script ( [fprime_gds_client_sample_script.py](https://github.com/fprime-community/fprime-gds-client-sample-script/blob/main/fprime_gds_client_sample_script.py) ) that filters for a specific channel, but could be extended. You could then use the script manually (easier) or make it a plugin (more implementation work, but then it starts up automatically with the GDS).
